### ISSUE TITLE
CLI No module named xxx when running uvicorn main:APP . --reload, but Pycharm ok?

### ISSUE THREAD

**Issue by scheung38** [tiangolo/fastapi#560]
**Description**

This is my file structure:

proj_root
   /rest_api/main.py

$ cd rest_api

$ uvicorn rest_api:APP --reload

results in one of the files 

ModuleNotFoundError: No module named xxxx

But when Right-Click rest_api.py in Pycharm it runs ok:

  INFO: Started server process [14928]
  INFO: Waiting for application startup.
  INFO: Uvicorn running on http://0.0.0.0:5000 (Press CTRL-C to quit)

**Additional context**

**Comment by euri10** [comment]
Without a better description it's impossible to help

**Comment by dmontagu** [comment]
PyCharm has some capabilities to automatically activate the appropriate virtual environment, and adding appropriate paths to the PYTHONPATH. Try checking the value of `sys.path` (and `sys.executable`) when run in pycharm, and when run outside pycharm. They probably differ, and that would explain why the module is found in one case and not in the other.

**Comment by euri10** [comment]
Given the number of similar related questions this superb tutorial https://chrisyeh96.github.io/2017/08/08/definitive-guide-python-imports.html# should be added to the FAQ once there is one,!

**Comment by scheung38** [comment]
Hi @dmontagu, in a way they are both ran within the PyCharm context

Right Click on main.py within PyCharm runs

But

cd rest_api

main.py:

    from mymodel import GlobaldB 
    app = FastAPI()

  
$uvicorn main:app --reload

ModuleNotFoundError: No module named mymodel

**Comment by euri10** [comment]
I think you'll get better support on import issues on [stackoverflow](https://stackoverflow.com/), this is in no way related to FastAPI but it's a general python issue. the link provided above should help too.

**Comment by dmontagu** [comment]
Yes, pycharm will configure the path when you run, but not in the terminal. You need to set up your import path, or change how you are organizing/referencing your package-internal files. The links @euri10 posted should be helpful for learning about how this stuff works.

**Comment by scheung38** [comment]
The only variable I find within RUN config of PyCharm is:

PYTHONUNBUFFERED=1
Add content roots to PYTHONPATH checked
Add source roots to PYTHONPATH checked

**Comment by tiangolo** [comment]
If you do:

```bash
cd rest_api
```

You would have to do:

```bash
uvicorn main:app
```

See more in the docs here: https://fastapi.tiangolo.com/tutorial/first-steps/

And here: https://fastapi.tiangolo.com/tutorial/bigger-applications/

**Comment by requiemofthesouls** [comment]
Hi @scheung38.
This may help you.

```fish
cd rest_api
export PYTHONPATH=$PWD 
uvicorn main:app --reload
```

> The only variable I find within RUN config of PyCharm is:
> 
> PYTHONUNBUFFERED:1
> Add content roots to PYTHONPATH: checked
> Add source roots to PYTHONPATH: checked

**Comment by tiangolo** [comment]
@scheung38 were you able to solve your problem? If so, you can close the issue.

**Comment by danielahdez** [comment]
> 
> 
> **Description**
> 
> This is my file structure:
> 
> proj_root
> /rest_api/main.py
> 
> $ cd rest_api
> 
> $ uvicorn rest_api:APP --reload
> 
> results in one of the files
> 
> ModuleNotFoundError: No module named xxxx
> 
> But when Right-Click rest_api.py in Pycharm it runs ok:
> 
> INFO: Started server process [14928]
> INFO: Waiting for application startup.
> INFO: Uvicorn running on http://0.0.0.0:5000 (Press CTRL-C to quit)
> 
> **Additional context**
> Add any other context or screenshots about the feature request here.

I'm having the same problem ... did you solve it?

**Comment by tiangolo** [comment]
@danielahdez that's probably related to the Python environment you are using and where your files are being installed.

This post has some useful info on handling environments: https://snarky.ca/a-quick-and-dirty-guide-on-how-to-install-packages-for-python/

If after that you still have problems, please create a new issue so we can help you with your specific problem.
